### ISSUE TITLE
Support self signed certificates

### ISSUE THREAD

**Issue by mulbc** [rozpuszczalny/vscode-redmine#3]
Please support self signed certificates for the Redmine instance.
For my Redmine the certificate is actually signed by a trusted CA (which is in my Keychain) but your extension doesn't pick up the trust.

System: OSX 10.13.3
VSCode: 1.21.1

![extension vscode-redmine redmine 2018-03-20 14-19-47](https://user-images.githubusercontent.com/131446/37657015-13f58d2c-2c4a-11e8-935d-bb66492d3216.jpg)

Invoked by: `Redmine: List open issues assigned to me`

**Comment by rozpuszczalny** [comment]
Hi,
unfortunately there is no easy fix to this, due to constant list of CA certificates in Node.js (see https://github.com/nodejs/node/issues/4175). I've added parameter called `redmine.rejectUnauthorized`, which sets rejectUnauthorized in https request options. Set it to `false` when you want to use self-signed certificates. It isn't the best solution, but it works. I'll add this in the next minor version of this extension with some other features.

**Comment by AdamMajer** [comment]
NodeJS doesn't have a constant list of certificates for a long time. It allows overrides both at command line and programmatically . `node --help | grep CA`

There is actually an environmental option,

NODE_TLS_REJECT_UNAUTHORIZED
             When set to 0, TLS certificate validation is disabled.

**Comment by rozpuszczalny** [comment]
Indeed, I've missed information about `https.request()` options, which supports using `ca`, `cert` etc. (ref: [https.request()](https://nodejs.org/api/https.html#https_https_request_url_options_callback)). 

I can't use environment variables, because I have no power to control startup of node script (VSCode manages it).

I'm reopening this issue to provide CA path parameter, however I can't provide support for CA added in system keychain (if there is a way, please correct me or provide PR ;) ).

**Comment by AdamMajer** [comment]
NodeJS uses OpenSSL so it depends how node was compiled and what OS it's running on. On Linux if using system CA's, it would pick up that CA. On Windows, probably not. It's system dependent.

Keep in mind that the `ca` option replaces all certs
